testflinger_agent/agent.py

```diff
@@ -32,6 +32,11 @@ def __init__(self, client):
         self.set_state("waiting")
         self.advertised_queues = self.client.config.get("advertised_queues")
         self.advertised_images = self.client.config.get("advertised_images")
+        location = self.client.config.get("location")
```

Comment:
Let's ensure this one is an empty string if there's no location set, otherwise it will be None.